bundles/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/api/model/UniFiController.java

```diff
@@ -97,18 +109,31 @@ public void stop() throws UniFiException {
 
     public void login() throws UniFiException {
         UniFiControllerRequest<Void> req = newRequest(Void.class);
-        req.setPath("/api/login");
         req.setBodyParameter("username", username);
         req.setBodyParameter("password", password);
         // scurb: Changed strict = false to make blocking feature work
         req.setBodyParameter("strict", false);
         req.setBodyParameter("remember", false);
+        if (this.unifiOS) {
+            req.setAuthPath("/api/auth/login");
+        } else {
+            req.setAuthPath("/api/login");
+        }
         executeRequest(req);
+        this.csrfToken = req.getCsrfToken();
     }
 
     public void logout() throws UniFiException {
         UniFiControllerRequest<Void> req = newRequest(Void.class);
```

Comment:
Would it maybe make more sense to wrap/extend the UniFiControllerRequest. Now a lot of implementation leaks out, so there is now a separate method to set auth paths, CSRF tokens are handled from inside and outside.

I'm even doubting whether the handling inside the Request makes sense, because I do not see them being re-used so I doubt whether the post actions actually work as intended?

Reply:
I was thinking the same. I'm currently looking at the source, thinking of the best way to integrate this A/B logic.

Reply:
> I'm even doubting whether the handling inside the Request makes sense

It doesn't, my implementation has it inside the Controller.